substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/BreakpointInterceptor.java

```diff
@@ -913,6 +918,119 @@ private static String asInternalSignature(Object paramTypesArray) {
+                                        checkedExceptionNames, modifiers, targetConstructorClassName));
+        Object result = false;
+        if (allowed) {
+            JNIValue args = StackValue.get(6, JNIValue.class);
+            args.addressOf(0).setObject(self);
+            args.addressOf(1).setObject(serializeTargetClass);
+            args.addressOf(2).setObject(parameterTypes);
+            args.addressOf(3).setObject(checkedExceptions);
+            args.addressOf(4).setInt(modifiers);
+            args.addressOf(5).setObject(targetConstructorClass);
+            result = nullHandle().notEqual(jniFunctions().getCallObjectMethodA().invoke(jni, bp.clazz, bp.method, args));
+            if (clearException(jni)) {
+                result = false;
+            }
+        }
+        JNIObjectHandle callerClass = getDirectCallerClass();
+        if (traceWriter != null) {
+            traceWriter.traceCall("serialization",
+                            "generateSerializationConstructor",
+                            null,
+                            null,
+                            null,
+                            result,
+                            serializeTargetClassName, parameterTypeNames,
+                            checkedExceptionNames, modifiers, targetConstructorClassName);
+            JNIFunctionPointerTypes.CallIntMethodFunctionPointer noArgRetIntCall = jniFunctions().getCallIntMethod();
+            int privateStaticFinalMask = Modifier.PRIVATE | Modifier.STATIC | Modifier.FINAL;
+            int staticFinalMask = Modifier.STATIC | Modifier.FINAL;
+
+            // call serializationTargetClass.getDeclaredFields();
+            JNIObjectHandle javaLangClass = agent.handles().findClass(jni, "java/lang/Class");
+            JNIMethodId getDeclaredFieldsMI = agent.handles().getMethodId(jni, javaLangClass, "getDeclaredFields",
+                            "()[Ljava/lang/reflect/Field;", false);
+            JNIFunctionPointerTypes.CallObjectMethod0FunctionPointer noArgRetObjectCall = jniFunctions().getCallObjectMethod();
+            JNIObjectHandle fieldsJArray = noArgRetObjectCall.invoke(jni, serializeTargetClass, getDeclaredFieldsMI);
+
+            // Prepare JNIMethodIds for later calls
+            JNIObjectHandle javaLangReflectField = agent.handles().findClass(jni, "java/lang/reflect/Field");
+            JNIMethodId getFieldNameId = agent.handles().getMethodId(jni, javaLangReflectField, "getName", "()Ljava/lang/String;", false);
+            JNIMethodId getFieldModifiersId = agent.handles().getMethodId(jni, javaLangReflectField, "getModifiers", "()I", false);
+            JNIMethodId getFieldTypeId = agent.handles().getMethodId(jni, javaLangReflectField, "getType", "()Ljava/lang/Class;", false);
+            // Add serialize and deserialize fields into reflection configs
+            // Check each field
```

Comment:
It would be cleaner to pass along the field information gathered below in the serialization event above, or in separate serialization events, rather than faking reflection accesses. The `TypeConfiguration` for reflection can be passed to the `SerializationProcessor` constructor, which would then be able to register the fields and classes accordingly.

Reply:
On second thought, is it even necessary to examine the fields here in the agent? It seems more straightforward to just do that during the image build for all classes in the serialization configuration and register their fields for reflective access then. That way, the reflective access to these fields is considered an implementation detail that does not need to be part of the reflection configuration files (unless they would be traced anyway).

Reply:
Agreed, I can move this part to build time. That would be cleaner and less overhead for agent.